### PR TITLE
JBPM-4625 Generation of JPA enabled models

### PR DESCRIPTION
**Pull request by wmedvede**
The following PR contains

1) persistence descriptor service for managing the persistence.xml file
2) persistence descriptor editor
3) persistence descriptor editor service

The commit includes not ready to production features but an iteration

### CODE REVIEW

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/main/java/org/kie/workbench/common/screens/datamodeller/backend/server/DataModelerEventObserver.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    }
+
+    @Inject
+    public DataModelerEventObserver( final PersistenceDescriptorService descriptorService,
+            final @Named( "ioStrategy" ) IOService ioService ) {
+        this.descriptorService = descriptorService;
+        this.ioService = ioService;
+    }
+
+    private void onDataObjectCreated( @Observes DataObjectCreatedEvent event ) {
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/main/java/org/kie/workbench/common/screens/datamodeller/backend/server/DataModelerEventObserver.java]: You've commented out invocation of the ```DataObjectCreatedEvent``` in ```DataModelBrowser```. Do you raise the event elsewhere too?

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/main/java/org/kie/workbench/common/screens/datamodeller/backend/server/DataModelerEventObserver.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                   descriptorService.save( descriptorPath,
+                           persistenceDescriptor,
+                           null,
+                           "Entity added to persistence descriptor" );
+               }
+            }
+        }
+    }
+
+    private void onDataObjectDeleted( @Observes DataObjectDeletedEvent event ) {
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/main/java/org/kie/workbench/common/screens/datamodeller/backend/server/DataModelerEventObserver.java]: You've commented out invocation of the ```DataObjectDeletedEvent``` in ```DataModelBrowser```. Do you raise the event elsewhere too?

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-api/src/main/java/org/kie/workbench/common/screens/datamodeller/service/PersistenceDescriptorService.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import org.uberfire.ext.editor.commons.service.support.SupportsRead;
+import org.uberfire.ext.editor.commons.service.support.SupportsUpdate;
+
+@Remote
+public interface PersistenceDescriptorService extends ViewSourceService<PersistenceDescriptorModel>,
+        ValidationService<PersistenceDescriptorModel>,
+        SupportsRead<PersistenceDescriptorModel>,
+        SupportsUpdate<PersistenceDescriptorModel, Metadata> {
+
+    PersistenceDescriptorModel createProjectDefaultDescriptor( final Path path );
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-api/src/main/java/org/kie/workbench/common/screens/datamodeller/service/PersistenceDescriptorService.java]: Would the ```Path``` parameter be better suited as ```Project```? The path for ```persistence.xml``` is always relative to a ```Project``` -- and your implementation resolves the ```Project``` from this ```Path``` anyway (although I can appreciate you probably don't have access to ```Project``` in the client-side code -- unless you use ```org.guvnor.common.services.project.context.ProjectContext```

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/main/java/org/kie/workbench/common/screens/datamodeller/backend/server/DataModelerEventObserver.java (hunk 1 above)
Comment (wmedvede) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/main/java/org/kie/workbench/common/screens/datamodeller/backend/server/DataModelerEventObserver.java]: NewDataObjectPopup is an old thinig that will be removed in next clean-up iteration.
The event is now thrown in the service

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/main/java/org/kie/workbench/common/screens/datamodeller/backend/server/DataModelerEventObserver.java (hunk 2 above)
Comment (wmedvede) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/main/java/org/kie/workbench/common/screens/datamodeller/backend/server/DataModelerEventObserver.java]: same as dataobject popup.

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-api/src/main/java/org/kie/workbench/common/screens/datamodeller/service/PersistenceDescriptorService.java (hunk 3 above)
Comment (wmedvede) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-api/src/main/java/org/kie/workbench/common/screens/datamodeller/service/PersistenceDescriptorService.java]: this implementation may probably change. Ideally the persistence.xml file will be created when the user selects to enable persistence for a given project. This involves an update on the project editor screens. But now I'm just launching the persistence descriptor editor.

**Comment by wmedvede** [comment]
To not enter in a never ending discussion I've sent part of previous work to the transh, and moved everithing to kie-wb-commons

**Pull request closed without merging**